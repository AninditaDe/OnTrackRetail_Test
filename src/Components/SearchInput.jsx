import React, { useState } from "react";
import { Col } from "react-bootstrap";

const SearchInput = ({ filter, setFilter }) => {
  const [search, setSearch] = useState(filter);

  return (
    <Col md="6">
      <input
        className="form-control rounded-pill block-example border border-info"
        type="text"
        placeholder="Search..."
        onChange={(e) => setSearch(e.target.value)}
        onBlur={() => setFilter(search)}
        value={search}
      />
    </Col>
  );
};

export default SearchInput;
