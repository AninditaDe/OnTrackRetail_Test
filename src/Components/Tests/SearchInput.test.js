import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SearchInput from "../SearchInput";

configure({ adapter: new Adapter() });

/* Can be extended */
describe("<SearchInput />", () => {
  it("render correctly text component <SearchInput/>", () => {
    const TextInputComponent = shallow(<SearchInput />);
    expect(TextInputComponent).toMatchSnapshot();
  });
});
