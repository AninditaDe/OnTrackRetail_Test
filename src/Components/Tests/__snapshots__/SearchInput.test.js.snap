// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> Should create an entry in component <SearchInput/> 1`] = `ShallowWrapper {}`;

exports[`<SearchInput /> render correctly text component <SearchInput/> 1`] = `ShallowWrapper {}`;
